### PR TITLE
Fix WebGL2 build with -sWASM_BIGINT build mode

### PR DESCRIPTION
**Pull request by juj**
Fix WebGL2 build with -sWASM_BIGINT build mode, where glClientWaitSync() and glWaitSync() will not take a low,high i32 pair as argument

### CODE REVIEW

File: src/library_webgl2.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ var LibraryWebGL2 = {
   },
 
   glGetInteger64v__sig: 'vii',
-  glGetInteger64v__deps: ['$emscriptenWebGLGet', '$writeI53ToI64'],
+  glGetInteger64v__deps: ['$emscriptenWebGLGet'],
```
Comment (juj) [inline src/library_webgl2.js:53]: Drive-by fix to remove a deps item that was out of date.

File: src/library_webgl2.js [hunk 2]
```diff
@@ -723,6 +723,22 @@ var LibraryWebGL2 = {
     GL.syncs[id] = null;
   },
 
+#if WASM_BIGINT
+  glClientWaitSync__sig: 'iiii',
+  glClientWaitSync: function(sync, flags, timeout) {
```
Comment (kripken) [inline src/library_webgl2.js:728]: I think this kind of thing can be handled without duplicating the functions, by using `{{{ defineI64Param('timeout') }}}` and related functions to read from the parameter or parameter pair.

Comment (kripken) [reply src/library_webgl2.js:728]: (grep for `defineI64Param` for examples)

Comment (juj) [reply src/library_webgl2.js]: Updated to avoid function duplication. Unfortunately the existing functions for parameter pairs are not fully adequate for this purpose, though `{{{ defineI64Param('timeout') }}}` works well.

File: src/library_webgl2.js [hunk 3]
```diff
@@ -723,21 +723,41 @@ var LibraryWebGL2 = {
...
     // In JS, there's no 64-bit value types, so instead timeout is taken to be signed, and GL_TIMEOUT_IGNORED is given value -1.
     // Inherently the value accepted in the timeout is lossy, and can't take in arbitrary u64 bit pattern (but most likely doesn't matter)
     // See https://www.khronos.org/registry/webgl/specs/latest/2.0/#5.15
-    return GLctx.clientWaitSync(GL.syncs[sync], flags, convertI32PairToI53(timeoutLo, timeoutHi));
+    return GLctx.clientWaitSync(GL.syncs[sync], flags,
+#if WASM_BIGINT
+      Number(timeout)
+#else
+      convertI32PairToI53(timeout_low, timeout_high)
+#endif
```
Comment (kripken) [inline src/library_webgl2.js:742]: This can be replaced with `{{{ receiveI64ParamAsI53('timeout', Infinity) }}}`?

Comment (kripken) [reply src/library_webgl2.js:742]: (Or, if I'm missing something, we could add a new helper function for this pattern, as it seems like it should be common?)

File: src/library_webgl2.js [hunk 4]
```diff
@@ -723,21 +723,41 @@ var LibraryWebGL2 = {
     GL.syncs[id] = null;
   },
 
+#if WASM_BIGINT
+  glClientWaitSync__sig: 'iiii',
```
Comment (sbc100) [inline src/library_webgl2.js:727]: I think you can just use `iiij` and avoid the different signatures.

File: src/library_webgl2.js (hunk 3 above)
Comment (sbc100) [reply src/library_webgl2.js:742]: Yes, see the definition of `fd_pwrite` in `src/library_wasi.js` for how we do this elsewhere.

File: src/library_webgl2.js [hunk 5]
```diff
@@ -723,21 +723,41 @@ var LibraryWebGL2 = {
...
+#else
   glClientWaitSync__sig: 'iiiii',
   glClientWaitSync__deps: ['$convertI32PairToI53'],
```
Comment (sbc100) [inline src/library_webgl2.js:730]: You can do `glClientWaitSync__deps: i53ConversionDeps,` and avoid the if/else block.

File: src/library_webgl2.js (hunk 4 above)
Comment (juj) [reply src/library_webgl2.js]: Ah right, forgot about that one. That works well.

File: src/library_webgl2.js (hunk 3 above)
Comment (juj) [reply src/library_webgl2.js:740]: Unofrunately `receiveI64ParamAsI53` emits a check for `bigintToI53Checked` or a `convertI32PairToI53Checked`, followed by a IsNaN, which are both undesirable in this case. The desired code is to call `convertI32PairToI53` like show in the PR.

File: src/library_webgl2.js (hunk 5 above)
Comment (juj) [reply src/library_webgl2.js:728]: Unfortunately that will add a dependency to `bigintToI53Checked` or `convertI32PairToI53Checked`, but in this case the function `convertI32PairToI53` is expected.

File: src/library_webgl2.js (hunk 3 above)
Comment (sbc100) [reply src/library_webgl2.js:740]: Why you explain why?  (i.e. why is convertI32PairToI53Checked undesirable here?)   It seems sad to have these generic mechanism and then not be able to use them.

Comment (juj) [reply src/library_webgl2.js:740]: The values in this case are timeouts, so if user specifies a long timeout, we are very ok to round here, and not abort on if rounding occurred. Also code size.

Comment (sbc100) [reply src/library_webgl2.js:740]: According to my math 2^^53 nanoseconds is about 104 days.    If we cap the values here doesn't that just mean that we don't support timeout longer than 104 days?   If that was the only downside it doesn't seem bad, and keeping the code simple might be better?

Or are there some magic values above 2^^53 that we need to support?

Comment (juj) [reply src/library_webgl2.js:740]: That is true. I don't know of any magic values, but I would like to stick to this smaller code size form, since in this case expected rounding and smaller code size align with each other (rounding is desired).

Comment (juj) [reply src/library_webgl2.js]: Ok, added a new function `receiveI64ParamAsI53Unchecked` now, how does that look?
